### PR TITLE
Integrate gateway registry with nym-wireguard

### PR DESCRIPTION
**Pull request by durch**
# Description

Integrate gateway registry with nym-wireguard, extract common types.

Related: NC-54

### CODE REVIEW

**Comment by durch** [comment]
Rebased on the latest develop

File: common/types/src/lib.rs [hunk 1]
```diff
@@ -9,8 +9,10 @@ pub mod error;
 pub mod fees;
 pub mod gas;
 pub mod gateway;
+pub mod gateway_client_registration;
 pub mod helpers;
 pub mod mixnode;
 pub mod pending_events;
 pub mod transaction;
 pub mod vesting;
+pub mod wireguard;
```
Comment (octol) [inline common/types/src/lib.rs:20]: I wonder if we should place all the wireguard stuff in this crate behind the `wireguard` feature toggle? Then we don't have to worry about nym-connect and the wallet pulling in wireguard

Comment (durch) [reply common/types/src/lib.rs:20]: This is the `PeerPublicKey` type, so its neater this way, I'll put it behind a feature flag tough

Comment (durch) [reply common/types/src/lib.rs:20]: Practical problem was that there was a circular dependency between gateway and wireguard, so the types got extracted out

**Review by octol — APPROVED** [review]

**Pull request merged**